app/api/groups/update_permissions.go

```diff
@@ -127,6 +111,39 @@ func (srv *Service) updatePermissions(w http.ResponseWriter, r *http.Request) se
 	return service.NoError
 }
 
+func checkAccessRightsNeededToUpdatePermissions(s *database.DataStore, user *database.User,
+	sourceGroupID, groupID, itemID int64) service.APIError {
+	// the authorized user should be a manager of the sourceGroupID and
+	// the 'sourceGroupID' should be an ancestor of 'groupID'
+	found, err := s.Groups().ManagedBy(user).Where("groups.id = ?", sourceGroupID).
+		Joins(`
+				JOIN groups_ancestors_active AS descendants
+					ON descendants.ancestor_group_id = groups.id AND descendants.child_group_id = ?`, groupID).
+		HasRows()
```

Comment:
The manager needs `can_grant_group_access` as well (on any group ancestor he is manager of)

Reply:
fixed